### PR TITLE
[SPARK-7482][SparkR] Rename some DataFrame API methods in SparkR to match their counterparts in Scala.

### PR DESCRIPTION
**Pull request by sun-rui**

### CODE REVIEW

**Comment by shivaram** [comment]
@sun-rui - Thanks for the PR. However I am having second thoughts about some of the renames. For some functions like `filter` and I guess to a certain extent `sample` its probably fine to overload as they are less often used functions.

However `load` and `save` are probably more widely used functions and our functions actually do something different from the `load` and `save` in R. One idea might be to rename these to `read` or `read.df` as these functions typically load data from external formats (like read.csv or read.table etc.). Any thoughts ?

cc @rxin

**Comment by sun-rui** [comment]
@shivaram, I think it is OK to rename APIs to ones that R user are accustomed to, for example, read.df and write.df. It would be better if there are more feedback from R community.

**Comment by shivaram** [comment]
Yeah lets go with `read.df` and `write.df`for now - We can also just alias these to `loadDF` and `saveDF`. We should revisit this when we graduate SparkR from experimental status and see if we have any user feedback

**Comment by shivaram** [comment]
@sun-rui Could you bring this up to date with master ?

**Comment by shivaram** [comment]
Jenkins, retest this please

File: R/pkg/R/SQLContext.R [hunk 1]
```diff
@@ -446,14 +446,15 @@ dropTempTable <- function(sqlCtx, tableName) {
...
 #' @examples
 #'\dontrun{
 #' sc <- sparkR.init()
 #' sqlCtx <- sparkRSQL.init(sc)
-#' df <- load(sqlCtx, "path/to/file.json", source = "json")
+#' df <- read.df(sqlCtx, "path/to/file.json", source = "json")
 #' }
 
-loadDF <- function(sqlCtx, path = NULL, source = NULL, ...) {
+read.df <- loadDF <- function(sqlCtx, path = NULL, source = NULL, ...) {
```
Comment (shivaram) [inline R/pkg/R/SQLContext.R]: I didn't know this was a valid way to declare alias functions in R. Hmm - But it seems to go against the style in the rest of the file. Could we just declare a new function `read.df` similar to other aliases ?

File: R/pkg/R/generics.R [hunk 2]
```diff
@@ -453,7 +453,11 @@ setGeneric("saveAsTable", function(df, tableName, source, mode, ...) {
   standardGeneric("saveAsTable")
 })
 
-#' @rdname saveAsTable
+#' @rdname write.df
+#' @export
+setGeneric("write.df", function(df, path, source, mode, ...) { standardGeneric("write.df") })
```
Comment (shivaram) [inline R/pkg/R/generics.R]: I guess we'll also need a generic for `read.df` ?

Comment (sun-rui) [inline R/pkg/R/generics.R]: No. read.df is not a generic function.

File: R/pkg/R/SQLContext.R (hunk 1 above)
Comment (sun-rui) [inline R/pkg/R/SQLContext.R]: fixed.

**Comment by sun-rui** [comment]
rebased to master.

**Comment by shivaram** [comment]
Thanks for the update. LGTM. Will wait for Jenkins and then merge

**Pull request closed without merging**